test/controller.go

```diff
@@ -0,0 +1,95 @@
+/*
```

Comment:
up to @aaron-prindle  - it's probably easier to add everything at once here, then remove in a subsequent PR

Reply:
It would make more sense to update these in a separate PR as removing the test/* and config/samples/* directory are both used in make test.  If those are removed, `make test` won't pass

Reply:
works.